File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        
        sort(intervals.begin(), intervals.end(),
             [](vector<int>& a, vector<int>& b) {
                 return a[1] < b[1];
             });


        int pe = intervals[0][1];
        int re = 0;

        for(int i = 1 ; i < intervals.size() ; i++){

            int start = intervals[i][0];
            int end = intervals[i][1];

            if(start < pe){
                re++;
            }
            else{
                pe = end;
            }

        }

        return re;

    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: list[list[int]]) -> int:

        intervals.sort(key=lambda x: x[1])

        pe = intervals[0][1]
        re = 0

        for i in range(1,len(intervals)):

            start = intervals[i][0]
            end = intervals[i][1]

            if start < pe:
                re += 1
            else:
                pe = end

        return re        

